syntax="proto3";

package accounts;

service AccountsService {
    rpc ListAccounts(UserRequest) returns (AccountsResponse);
    rpc UpdateBalance(UpdateBalanceRequest) returns (UpdateBalanceResponse);
}

message Account {
    int32 id = 1;
    string user_id = 2;
    double balance = 3;
}

message UserRequest {
    string user_id = 1;
}

message AccountsResponse {
    repeated Account accounts = 1;
}

message UpdateBalanceRequest {
    int32 account_id = 1;
    double amount = 2;
}

message UpdateBalanceResponse {
    bool success = 1;
    double new_balance = 2;
    string message = 3;
    string user_id = 4;
}